nameable.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator < Nameable
```

Comment:
- [OPTIONAL] To be able to read from and/or write to the `nameable` instance object, you will need to define both ` setter` and `getter` methods like(for example) so: `attr_accessor :nameable`, please kindly fix that.